### PR TITLE
Add time-varying greenhouse gases to RRTMG shortwave options

### PR DESCRIPTION
**Pull request by weiwangncar**
TYPE: enhancement

KEYWORDS: time-varying greenhouse gases, RRTMG-sw

SOURCE: internal

DESCRIPTION OF CHANGES:
Problem:
Time-varying greenhouse gases were added to the WRF code in 2013. It was added to CAM longwave and shortwave options. But it was only added to RRTMG longwave scheme. 

Solution:
1. After consulting with the original contributors of the code, the time-varying gases are added to the RRTMG-sw and RRTMG-swf code in this PR. 
2. Also included in this PR is to remove dependency of clwrf_support code for RRTMG-lwk (KIM version) - that code does not use the capability. 

In a single 15 km domain test in July 2018, there are small differences between temperature and height at 24 h forecast time as shown below in the domain averaged profile plot:
diff2

LIST OF MODIFIED FILES: 
M       main/depend.common
M       phys/module_ra_rrtmg_sw.F
M       phys/module_ra_rrtmg_swf.F

TESTS CONDUCTED: 
1. Time varying gases are added to RRTMG-swf and RRTMG-sw
2. The Jenfins tests all passing.

RELEASE NOTE: Added time-varying greenhouse gases to RRTMG shortwave options: RRTMG and RRTMG-fast.

### CODE REVIEW

**Comment by weiwangncar** [comment]
Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           59
    Number of Simulations  : 159           157        0
    Number of Comparisons  : 96           95        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

File: phys/module_ra_rrtmg_sw.F [hunk 1]
```diff
@@ -10251,6 +10252,10 @@ SUBROUTINE RRTMG_SWRAD(                                        &
     integer:: idx_rei
     real:: corr
 
+#ifdef CLWRFGHG
+! Using data from CAMtr_volume_mixing_ratio data file
+    REAL(8)                                      :: co2, n2o, ch4, cfc11, cfc12
```
Comment (davegill) [inline phys/module_ra_rrtmg_sw.F:10257]: @weiwangncar 
Wei,
Here and the identical code in swf:
Replace
```
REAL(8)
```
with
```
REAL(KIND=8)
```

Comment (weiwangncar) [reply phys/module_ra_rrtmg_sw.F:10257]: @davegill The code in module_ra_rrtmg_lw.F uses real(8). We may need to change that later.

Comment (davegill) [reply phys/module_ra_rrtmg_sw.F]: @weiwangncar 
Wei,
No probs at all to only do some of this. We can't fix all questionable syntax, but when we come to it, we can clean it up. To me, it is fine to leave the other code AS-IS.

**Review by davegill — DISMISSED** [review]

**Comment by davegill** [comment]
@weiwangncar 
Wei,
I am OK with this PR. I approved it.

For a bigger picture, what does this PR do for the v4.4 release topic of time dependent gases?

**Comment by weiwangncar** [comment]
@davegill Thanks. Currently using time-varying greenhouse gases (not just co2) is a compile option (add -DCLWRFGHG). This PR doesn't change that. This PR only adds the gases to the shortwave code. Ming is working on making this a runtime option for v4.4.

**Comment by dudhia** [comment]
Is the plot height or geopotential height? Make sure before merging.

File: main/depend.common [hunk 2]
```diff
@@ -497,9 +497,7 @@ module_ra_rrtmg_lw.o: ../share/module_model_constants.o \
 		 module_ra_clWRF_support.o 
 module_ra_rrtmg_lwf.o: ../share/module_model_constants.o \
 		 module_ra_clWRF_support.o 
-module_ra_rrtmg_lwk.o: ../share/module_model_constants.o \
-                 module_ra_clWRF_support.o
-
+module_ra_rrtmg_lwk.o: ../share/module_model_constants.o
```
Comment (dudhia) [inline main/depend.common:500]: why does lwk not need clWRF_support?

Comment (weiwangncar) [reply main/depend.common:500]: @dudhia The GHG code is not in lwk.

**Comment by dudhia** [comment]
I deleted the comment. lwk is not lwf.

**Comment by davegill** [comment]
@weiwangncar @dudhia 
> There may be some advantage to add this PR to 4.3.2. Since 4.3 code has been around for a while, users could choose to use this instead of the new 4.4 code. If they do, they could have this fix to use.

Umm, I just sent out a big note saying no more v4.3.2 mods, and our emails were pointing in the direction of this being on the develop branch. Can this stay on the develop branch? For power users, they can always check out the develop branch for this mod.

**Review by dudhia — DISMISSED** [review]
I think it is OK for 4.3.2 or develop. Others can decide.

**Review by davegill — APPROVED** [review]

**Review by dudhia — APPROVED** [review]

**Comment by davegill** [comment]
@weiwangncar @dudhia @kkeene44 
Wei,
This compile-time RRTMG sw and swf option for GHG is now in the release-v4.3.2 branch. Kelly waited until we were finished changing the base, so users will now have this option available.

**Comment by weiwangncar** [comment]
@chenming You may want to include this in your testing and code change for making CLWRFGHG a runtime option.

**Comment by davegill** [comment]
@weiwangncar @smileMchen @kkeene44 
> You may want to include this in your testing and code change for making CLWRFGHG a runtime option.

Folks,
GHG mods went to the v4.3.2 branch. Once Kelly also pushes those mods to the develop branch, THEN Ming can use the corrected LW and SW mods for CAM, RRTMG, and RRTMG(f) natively on the develop branch as a place to start. I agree with Wei to a) use those mods, and b) wait for those mods.

**Pull request merged**